### PR TITLE
[FEQ]Yaswanth/FEQ-1980 Added test cases for toggle switch component

### PR DESCRIPTION
**Pull request by yaswanth-deriv**
changes:
- refactored `ToggleSwitch`:

  1. added internal state for `isChecked` . its default value can be controlled from outside using `value` props.
  2. added internal `ClickHandler` to control the state internally. you can still pass `onChange` if you need to perform further actions when switching

- Added test cases for toggle switch component in deriv-com/ui library

### CODE REVIEW

File: src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { ToggleSwitch } from '..';
+
+describe('ToggleSwitch Component', () => {
+
+    it('handles onChange event correctly', () => {
```
Comment (shayan-deriv) [inline src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx:7]:
```suggestion
    it('should render and function properly with default Props', () => {
```

File: src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        const onChange = jest.fn();
+        const { getByRole } = render(<ToggleSwitch onChange={onChange} value={false} />);
+        const toggleSwitch = getByRole('checkbox');
+        fireEvent.click(toggleSwitch);
+        expect(onChange).toHaveBeenCalledTimes(1);
```
[the remark this answers is not in the document]
Comment (yaswanth-deriv) [reply src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx:12]: I have added these cases by using props.I will add now by using fire event.Thanks

File: stories/Badge.stories.tsx [hunk 3]
```diff
@@ -81,7 +81,7 @@ export const ContainedPurpleRounded: Story = {
 };
 
 export const ContainedRed: Story = {
-    name: "Contained (Yellow)",
+    name: "Contained (Red)",
```
Comment (shayan-deriv) [inline stories/Badge.stories.tsx:84]: @yaswanth-deriv since this is irrelevant, can we fix it in separate pr?

File: src/components/ToggleSwitch/index.tsx
Comment (shayan-deriv) [inline src/components/ToggleSwitch/index.tsx:1]: @yaswanth-deriv I refactored the Component. please check it 🙏

File: src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx
Comment (shayan-deriv) [inline src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx:1]: @yaswanth-deriv I added some changes based on the new implementation. it's cleaner now. please check it 🙏

**Review by yaswanth-deriv — COMMENTED** [review]
LGTM

File: src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx
Comment (yaswanth-deriv) [reply src/components/ToggleSwitch/__test__/ToggleSwitch.spec.tsx:1]: I have checked it.It is fine.Please merge this PR

**Review by shayan-deriv — APPROVED** [review]

**Pull request merged**